rmc/utils/mpc.py

```diff
@@ -398,100 +404,123 @@ def _run_glm(
+            model_aics.append(model.aic)
+            model_formulas.append(formula)
+
+        # Find model with lowest AIC
+        min_model_aic = min(model_aics)
+        min_model_idx = model_aics.index(min_model_aic)
+        min_model_var = var_combs[min_model_idx]
+        min_model = models[min_model_var]
+        min_model_formula = model_formulas[min_model_idx]
+        if model_aics.count(min_model_aic) > 1:
+            logger.warning(
+                """
+                There is a tie for minimum AIC in the %s regressions.
+                This function will use the first variable combination it finds by default!
+                """,
+                log_model_type,
+            )
+        logger.info(
+            "Model with smallest AIC for %s regressions used variables: %s",
+            log_model_type,
+            min_model_var,
         )
-    logger.info(
-        "Model with smallest AIC for single variable regressions used %s",
-        min_single_aic_var,
-    )
 
-    logger.info("Running joint (additive interactions only) regression...")
-    add_formula = f"pop_v_path ~ {' + '.join(variables)}"
-    add_X, add_model = _run_glm(add_formula)
+        return (min_model, min_model_formula)
 
-    logger.info("Running joint regression with all interactions...")
-    mult_formula = f"pop_v_path ~ {' * '.join(variables)}"
-    mult_X, mult_model = _run_glm(mult_formula)
+    all_vars = variables + additional_variables
+    min_models = {}
+    min_model_formulas = {}
+    min_model_aics = {}
+
+    # Find model with lowest AIC for single variable regressions
+    min_models["single"], min_model_formulas["single"] = _get_min_aic_glm(
+        [x for x in all_vars]
+    )
+
+    # List possible variable combinations for joint regressions
+    var_combs = []
+    for k in range(2, len(all_vars) + 1):
+        var_combs += [x for x in combinations(all_vars, k)]
+    # Find models with lowest AICs for joint variable regressions (with or without interactions)
+    for model_type in ["additive", "multiplicative"]:
+        min_models[model_type], min_model_formulas[model_type] = _get_min_aic_glm(
+            var_combs=var_combs, model_type=model_type
+        )
 
     logger.info("Running joint regression with specific interactions...")
     # Currently hardcoded to be formula from ExAC
-    spec_formula = "pop_v_path ~ oe + misbad + oe:misbad + polyphen + oe:polyphen"
-    spec_X, spec_model = _run_glm(spec_formula)
-
-    all_model_aic = single_var_aic + [add_model.aic, mult_model.aic, spec_model.aic]
-    min_aic = min(all_model_aic)
-    logger.info("Lowest model AIC: %f", min_aic)
-    if all_model_aic.count(min_aic) > 1:
+    min_model_formulas[
+        "spec"
+    ] = "pop_v_path ~ oe + misbad + oe:misbad + polyphen + oe:polyphen"
+    min_models["spec"] = _run_glm(min_model_formulas["spec"])
+
+    min_model_aics = {x: min_models[x].aic for x in model_types}
+    overall_min_aic = min(min_model_aics)
```

Comment:
```suggestion
    overall_min_aic = min(min_model_aics.values())
```
considering `min_model_aics` is a dictionary